exchange/utils.go

```diff
@@ -231,6 +257,49 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 	return
 }
 
+// cloneBidderReq - clones bidder request and replaces req.User and req.Device with new copies
+func cloneBidderReq(req *openrtb2.BidRequest) *openrtb_ext.RequestWrapper {
+
+	// bidder request may be modified differently per bidder based on privacy configs
+	// new request should be created for each bidder request
+	// pointer fields like User and Device should be cloned and set back to the request copy
+	var newReq *openrtb2.BidRequest
+	newReq = ptrutil.Clone(req)
+
+	if req.User != nil {
+		var userCopy *openrtb2.User
+		userCopy = ptrutil.Clone(req.User)
```

Comment:
Can these lines be rewritten as `userCopy := ptrutil.Clone(req.User)`. I see a similar pattern throughout this function as well.

Reply:
Rewritten to `userCopy := ortb.CloneUser(req.User)`
I didn't know this function exists.